src/substrate/subscribeFunc.ts

```diff
@@ -104,24 +103,37 @@ export const subscribeEvents: SubscribeFunc<ApiPromise, Block, ISubscribeOptions
         && (!offlineRange || !offlineRange.startBlock || offlineRange.startBlock < lastBlockNumber)) {
       offlineRange = { startBlock: lastBlockNumber };
     }
+    // if running in archival mode and can't determine the offlineRange then start from Block 0
+    if(archival && (!offlineRange || !offlineRange.startBlock)){
+      offlineRange = {startBlock: 0}
+    }
 
     // if we can't figure out when the last block we saw was, do nothing
     // (i.e. don't try and fetch all events from block 0 onward)
-    if (!offlineRange || !offlineRange.startBlock) {
-      log.warn('Unable to determine offline time range.');
-      return;
-    }
-
-    // poll the missed blocks for events
-    try {
-      const blocks = await poller.poll(offlineRange);
-      await Promise.all(blocks.map(processBlockFn));
-    } catch (e) {
-      log.error(`Block polling failed after disconnect at block ${offlineRange.startBlock}`);
+    // if (!offlineRange || !offlineRange.startBlock) {
+    //   log.warn('Unable to determine offline time range.');
+    //   return;
+    // }
+
+    // if running in archival mode then fetch blocks from the chain in batches 
+    // from starting block to the head of the node
+    if(archival){
+      const CHUNK_SIZE = 500;
+      const header = await api.rpc.chain.getHeader();
+      offlineRange['endBlock'] =  +header.number;
+      for (let block = offlineRange.startBlock + CHUNK_SIZE; block <= offlineRange.endBlock; block += CHUNK_SIZE) {
+        // poll the missed blocks for events
+        try {
+        const blocks = await poller.poll({startBlock: block - CHUNK_SIZE, endBlock: Math.min(block, offlineRange.endBlock)}, CHUNK_SIZE);
+        await Promise.all(blocks.map(processBlockFn));
+        } catch (e) {
+          log.error(`Block polling failed after disconnect at block ${offlineRange.startBlock}`);
+        }
+      }
```

Comment:
No else branch? So we can't poll outside of archival mode anymore? This would seemingly break existing functionality.

Reply:
Rectified. Added an else block which executes poller normally.